processor/src/test/java/org/mapstruct/ap/test/bugs/_3678/Issue3678Mapper.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct.ap.test.bugs._3678;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.mapstruct.AfterMapping;
+import org.mapstruct.BeforeMapping;
+import org.mapstruct.Mapper;
+import org.mapstruct.ReportingPolicy;
+
+@Mapper(unmappedSourcePolicy = ReportingPolicy.ERROR)
```

Comment:
This is not needed for our tests as it'll anyways fails if there is a compiler warning